src/gui/options_imp.cpp

```diff
@@ -583,6 +587,12 @@ int options_imp::getProxyType() const
     }
 }
 
+void options_imp::setLimitStep(QSpinBox *spinLimit, int value) {
```

Comment:
I think this should be a pure function as
```c++
int options_imp::limitStep(int value) const
{
}
```

that means `spinLimit->setSingleStep()` logic should be moved up to `options_imp::adaptDownloadLimitStep` and the like.